app/lib/feed_manager.rb

```diff
@@ -161,13 +161,19 @@ def filter_from_mentions?(status, receiver_id)
   # either action is appropriate.
   def add_to_feed(timeline_type, account, status)
     timeline_key = key(timeline_type, account.id)
-    reblog_key = key(timeline_type, account.id, 'reblogs')
+    reblog_key   = key(timeline_type, account.id, 'reblogs')
 
     if status.reblog?
+      reblog_set_key = key(timeline_type, account.id, "reblogs:#{status.reblog_of_id}")
+
       # If the original status or a reblog of it is within
       # REBLOG_FALLOFF statuses from the top, do not re-insert it into
       # the feed
       rank = redis.zrevrank(timeline_key, status.reblog_of_id)
+
+      redis.sadd(reblog_set_key, status.reblog_of_id) unless rank.nil?
```

Comment:
Does this get us anything? I think it might be best to only put reblogs in the `reblogs` set: if the original status is deleted, the reblogs should go with it (and we should check for that case in `unpush`), but there's nothing obvious to be gained by tracking the fact that the original status was in the timeline at some point, is there?

Reply:
If the final *reblog* is deleted, but the original *was* originally in the feed, we want to re-insert the original - no?

Reply:
We don't remove the original when we see the reblog (and in fact, we ignore the reblog), so no, we probably don't need to do anything to track the old status, and should move the other `sadd` to below the `return false`.